packages/nitro-utils/src/rollupPlugins/wrapServerEntryWithDynamicImport.ts

```diff
@@ -0,0 +1,219 @@
+import { consoleSandbox, flatten } from '@sentry/utils';
+import type { InputPluginOption } from 'rollup';
+
+export const SENTRY_WRAPPED_ENTRY = '?sentry-query-wrapped-entry';
+export const SENTRY_WRAPPED_FUNCTIONS = '?sentry-query-wrapped-functions=';
+export const SENTRY_REEXPORTED_FUNCTIONS = '?sentry-query-reexported-functions=';
+export const QUERY_END_INDICATOR = 'SENTRY-QUERY-END';
+
+/**
+ * A Rollup plugin which wraps the server entry with a dynamic `import()`. This makes it possible to initialize Sentry first
+ * by using a regular `import` and load the server after that.
+ * This also works with serverless `handler` functions, as it re-exports the `handler`.
+ *
+ * @param config Configuration options for the Rollup Plugin
+ * @param config.serverConfigFileName Name of the Sentry server config (without file extension). E.g. 'sentry.server.config'
+ * @param config.resolvedServerConfigPath Resolved path of the Sentry server config (based on `src` directory)
+ * @param config.entryPointWrappedFunctions Exported bindings of the server entry file, which are wrapped as async function. E.g. ['default', 'handler', 'server']
+ * @param config.additionalImports Adds additional imports to the entry file. Can be e.g. 'import-in-the-middle/hook.mjs'
+ * @param config.debug Whether debug logs are enabled in the build time environment
+ */
+export function wrapServerEntryWithDynamicImport(config: {
+  serverConfigFileName: string;
+  resolvedServerConfigPath: string;
+  entrypointWrappedFunctions: string[];
+  additionalImports?: string[];
+  debug?: boolean;
```

Comment:
m: Could we define an exported type for this instead of inlining it? In case the consuming sides need to expose these via their own options, it would be prone to running out of sync.